test/common/index.js

```diff
@@ -690,12 +690,26 @@ Object.defineProperty(exports, 'hasIntl', {
 });
 
 // Useful for testing expected internal/error objects
+const kAcceptableErrorTypes = ['NodeError', 'AssertionError', 'RangeError'];
 exports.expectsError = function expectsError({code, type, message}) {
-  return function(error) {
+  const expectedTypeName = type && type.name;
+  return function expectsErrorCheck(error) {
     assert.strictEqual(error.code, code);
-    if (type !== undefined)
-      assert(error instanceof type,
-             `${error} is not the expected type ${type}`);
+    if (expectedTypeName) {
```

Comment:
Yes, the downcast case is covered, but the upcast is not (api that throw `Error` changed to throw `TypeError`, test still pass). I found [one such case](https://github.com/nodejs/node/pull/13686/commits/215099b06a3addffe33a21c78fa5f2e6e62a5e23#diff-9e1e157e684832463865e5433e9be5deL153).
You know what will happen, we'll add `strictType` then we'll deprecate `type` and add a lint rule...
If you look at the change-set of the PR now, you'll see that enforcing strict type needs no test changes, and will protect from `Error`s being accidently specialized.